.github/workflows/pull_request.yml

```diff
@@ -0,0 +1,127 @@
+# SPDX-FileCopyrightText: Copyright (c) 2022, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: Build pull request
+
+on:
+  push:
+    branches:
+      - 'pull-request/**'
+
+# This allows a subsequently queued workflow run to interrupt previous runs
+concurrency:
+  group: '${{ github.workflow }} @ ${{ github.event.pull_request.head.label || github.head_ref || github.ref }}'
+  cancel-in-progress: true
+
+env:
+  AWS_DEFAULT_REGION: us-west-2
+  AWS_ACCESS_KEY_ID: "${{ secrets.GHA_AWS_ACCESS_KEY_ID }}"
+  AWS_SECRET_ACCESS_KEY: "${{ secrets.GHA_AWS_SECRET_ACCESS_KEY }}"
+  CHANGE_TARGET: "${{ github.base_ref }}"
+  CUDA_PATH: "/usr/local/cuda/"
+  CUDA_VER: "11.5"
+  GH_TOKEN: "${{ github.token }}"
+  GIT_COMMIT: "${{ github.sha }}"
+  MORPHEUS_ROOT: "${{ github.workspace }}/morpheus"
+  WORKSPACE: "${{ github.workspace }}/morpheus"
+  WORKSPACE_TMP: "${{ github.workspace }}/tmp"
+
+
+jobs:
+  check:
+    name: Check
+    runs-on: [self-hosted, linux, amd64, cpu4]
+    timeout-minutes: 60
+    container:
+      image: rapidsai/ci:cuda11.5.1-ubuntu20.04-py3.8
+    strategy:
+      fail-fast: true
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          lfs: false
+          path: 'morpheus'
+          fetch-depth: 0
+
+      - name: Check
+        shell: bash
+        run: ./morpheus/ci/scripts/github/checks.sh
+
+  build:
+    name: Build
+    runs-on: [self-hosted, linux, amd64, cpu16]
+    timeout-minutes: 240
```

Comment:
This seems really long

Reply:
Good catch, thats a hold-over from when we built cudf ourselves